### PR TITLE
Add discovery row page

### PR DESCRIPTION
**Pull request by RubenSousaDinis**
## Summary

This PR adds the discovery row pages. We're adding them to showcase talent inside particular areas, such as web3, desing, development, etc. The pages should be public accessible.

## Notion link

https://talentprotocol.notion.site/Improve-Discover-Page-Automation-2645e05c601945d2b1cffa339d4c8d10

### CODE REVIEW

File: app/services/talents/search.rb [hunk 1]
```diff
@@ -16,17 +18,31 @@ def call
...
+
+      users = users.where(
+        "users.username ilike :keyword " \
+        "OR users.display_name ilike :keyword " \
+        "OR tokens.ticker ilike :keyword " \
+        "OR tags.description ilike :keyword ",
+        keyword: "%#{keyword}%"
+      )
+
+      talents.where(user: users.distinct.select(:id))
```
Comment (verdefred) [inline app/services/talents/search.rb:45]: what's the reason for changing this from a talent query to a user and a talent one?

File: app/controllers/discovery_controller.rb [hunk 2]
```diff
@@ -28,9 +29,36 @@ def index
...
+
+    respond_to do |format|
+      format.html
+      format.json {
+        render(
+          json: @talents,
+          status: :ok
+        )
+      }
+    end
```
Comment (verdefred) [inline app/controllers/discovery_controller.rb:56]: I think we don't need this, right?

File: app/services/talents/search.rb (hunk 1 above)
Comment (RubenSousaDinis) [reply app/services/talents/search.rb:45]: The distinct does not work with the setseed. This way I do the distinct query in the users query and treat the talent query as it was before.

File: app/blueprints/discovery_row_blueprint.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class DiscoveryRowBlueprint < Blueprinter::Base
-  fields :id, :title, :badge, :badge_link
+  fields :id, :badge, :badge_link, :slug, :title, :description, :logo_url, :talents_count
```
[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply app/blueprints/discovery_row_blueprint.rb:2]: I'm a bit against writing code that might be needed in the future. I will include them in the normal view for now and when we need another view we can refactor.

[the remark this answers is not in the document]
Comment (RubenSousaDinis) [reply app/blueprints/discovery_row_blueprint.rb]: Changed!

File: app/controllers/discovery_controller.rb (hunk 2 above)
Comment (RubenSousaDinis) [reply app/controllers/discovery_controller.rb:56]: In the discovery rows we can also search and we use this endpoint to also return the json results.

Comment (verdefred) [reply app/controllers/discovery_controller.rb:56]: makes sense, missed that. thanks!

File: app/packs/src/components/discovery/discovery_rows.jsx [hunk 4]
```diff
@@ -69,17 +73,17 @@ const DiscoveryRows = ({ discoveryRows, updateFollow }) => {
       return (
         <a
           href={row.badge_link}
-          className="cursor-pointer ml-2 "
+          className="cursor-pointer ml-2"
           target="_blank"
         >
-          <Tag className="bg-primary text-white cursor-pointer">
+          <Tag className=" cursor-pointer" mode={theme.mode()}>
```
Comment (verdefred) [inline app/packs/src/components/discovery/discovery_rows.jsx:79]: I don't think the tag component needs the mode anymore. it should work the same without it

Comment (RubenSousaDinis) [reply app/packs/src/components/discovery/discovery_rows.jsx]: removed!

**Review by verdefred — DISMISSED** [review]
Great work 🚀

**Review by verdefred — APPROVED** [review]

**Pull request merged**